zuul-core/src/main/java/com/netflix/zuul/filters/endpoint/ProxyEndpoint.java

```diff
@@ -389,11 +378,14 @@ private void storeAndLogOriginRequestInfo() {
         if (attemptToChosenHostMap == null) {
             attemptToChosenHostMap = new HashMap<>();
         }
-        String ipAddr = origin.getIpAddrFromServer(chosenServer.get());
-        if (ipAddr != null) {
-            attemptToIpAddressMap.put(attemptNum, ipAddr);
-            eventProps.put(CommonContextKeys.ZUUL_ORIGIN_ATTEMPT_IPADDR_MAP_KEY, attemptToIpAddressMap);
-            context.put(CommonContextKeys.ZUUL_ORIGIN_ATTEMPT_IPADDR_MAP_KEY, attemptToIpAddressMap);
+
+        if (chosenServer.get() != null) {
```

Comment:
When would this be null, and why didnt the previous code check for it?

Reply:
This can now happen because `computeReadTimeout` is intended to also validate conditions. For example, it could throw a timeout exception and thus no server will be chosen.

Reply:
Consider adding a comment, it's not obvious from reading the code that case can happen.